### DISCUSSION TITLE
近期发现可能有机构用机器人白嫖，各位小心

### DISCUSSION THREAD

**Discussion by MasterXuuu** [ChatGPTNextWeb/NextChat#2462] [General]
十分喜欢该项目，用该项目给自己和小伙伴部署了一个私有 GPT
为了小伙伴们的方便，没有设置密码，只是端口上用了奇怪的端口简单防护 
api 方面，为了稳定，付费用了一个代理的机构 https://aiproxy.io/，简单充了 200 ，准备可以用好久了
结果，今天突然小伙伴反馈，api 失效，显示余额不足
登录后台一看
<img width="1511" alt="image" src="https://github.com/Yidadaa/ChatGPT-Next-Web/assets/40448945/d49ee441-6e3f-4201-8ef5-a943af9070f4">
短时间被疯狂调用，榨干。
无奈，先给自己部署的项目页面换个地址端口，上 https，设置密码。
但转头还是想研究一下，于是在原有地址上，手搓了一个 python，看看对方机器人如何工作
<img width="967" alt="image" src="https://github.com/Yidadaa/ChatGPT-Next-Web/assets/40448945/0268dd70-fc9c-44a8-af4b-95c369fe86b1">
发现机器人会请求类似于
just say 'OK'
Just say this letter: 'T'
来试探各个模型是否能正确工作，然后后续会有真的问答请求，哈哈
那么我也来骚的吧，用 python 伪造一个真的 gpt，然后在请求问答内容的时候，骂一下他们

代码很丑，怎么快怎么来
<img width="1512" alt="image" src="https://github.com/Yidadaa/ChatGPT-Next-Web/assets/40448945/dd085614-353f-4fff-9e4e-e695f990cd6d">

[code.zip](https://github.com/Yidadaa/ChatGPT-Next-Web/files/12136978/code.zip)

已经部署到原地址了，目前有看到请求，哈哈哈

总之，各位小心，还是上个强一点的密码

**Comment by MasterXuuu** [comment]
部署，睡觉，明天看看情况

**Comment by Yidadaa** [comment]
不上密码就等于裸奔，认识的公益站都已经被薅秃了。

**Comment by gaboolic** [comment]
我记得之前不设置密码 在vercel里面部署不了的

**Comment by greatlsd** [comment]
有办法设置ip白名单吗？可以指定ip才能访问，就没有后顾之忧了
